MAXWELL HAMZAH

maxhamz opened the pull request:
1. Used model.Project as umbrella, since one project may have multiple todos (many2many)
2. EXPLORATION: Completed invitation method for members in database
3. EXPLORATION: assigned expiry time for token (1hr), with error handlers updated
4. PENDING: 
- ROCKET -> CRUD for todo using authorization (only projects member may crud the todos in one project)
- 3rd Party API: calendar (not Google Calendar), GeoLocation or AirVisual, considering relevance, versatility & documentation.
- CSS, frontend

arnoldtherigan15 commented on README.md:
```diff
@@ -1 +1,570 @@
+
+* **Success Response:**
+
+  * **Code:** 200 <br />
```
status code 201

arnoldtherigan15 commented on README.md:
```diff
@@ -1 +1,570 @@
+
+
+  * **Code:** 500 INTERNAL SERVER ERROR <br />
+    **Content:** <br>
```
error di handel ya pesannya bukan pesan yg panjang gitu

maxhamz replied:
berees

arnoldtherigan15 commented on controllers/project.js:
```diff
@@ -0,0 +1,277 @@
+class ProjectController {
+
+    static fetchProjects(req, res, next) {
+        console.log(">>> CONTROLLERS/PROJECT: FETCH PROJECTS");
```
kalau di kumpulin ke aku nantinya console.log2 nya di hapus aja ya

maxhamz replied:
berees

arnoldtherigan15 commented on controllers/project.js:
```diff
@@ -0,0 +1,277 @@
+    customError
+} = require("../helpers/customError.js")
+
+let user_id
```
ini variabel2 gak usah di buat global aja, di buat di masing2 method yg kamu pakai aja

maxhamz replied:
udah kusunat, sisa yg perlu2 bgt aja (yg bakalan dipake oleh sedikitnya 2 methods)

arnoldtherigan15 commented on controllers/project.js:
```diff
@@ -0,0 +1,277 @@
+                    include: [User, Project]
+                })
+            })
+            .then(response2 => {
```
ini gak masalah kalau kamu tulis response juga, gakusah nulis response2

maxhamz replied:
udah kuberesin...

arnoldtherigan15 commented on helpers/bcrypt.js:
```diff
@@ -0,0 +1,29 @@
+const salt = bcrypt.genSaltSync(10)
+
+function hashPassword(inputted) {
+    console.log("--- HELPERS: HASHING PASSWORD ---");
```
console2nya di hapus ya biar ga pusing

maxhamz replied:
berees

arnoldtherigan15 commented on helpers/bcrypt.js:
```diff
@@ -0,0 +1,29 @@
+    console.log(inputted);
+    console.log("STORED IS");
+    console.log(stored);
+    let flag = bcrypt.compareSync(inputted, stored)
```
biar lebih rapih sebenarnya gak perlu di simpan variabel, soalnya kamu ga pakai lagi variabel itu, langsung di return aja

maxhamz replied:
berees

arnoldtherigan15 commented on helpers/jwt.js:
```diff
@@ -0,0 +1,23 @@
+function verifyToken(logged) {
+    console.log(">>> ENTERING VERIFY-TOKEN <<<");
+    console.log("HEADS UP! FLAG IS");
+    let flag = jwt.verify(logged, process.env.SECRET)
```
di hapus aja flag ini gak di pakai

maxhamz replied:
berees

maxhamz commented:
UPDATE SERVER: SELASA
1. Removed console.log tests after function is ready
2. Completed CRUD for todos (project-member authorization) & projects (project-owner authorization)
3. Utilized geolocation as 3rd Party API requirement: find location by IP address, then add is as register & login parameter
4. Updated README's Error 500 display
5. Trimmed number of global variables, leaving only those which are used by AT LEAST 2 methods in a controller class.
5. PENDING:
- Client-side
- Google OAuth

maxhamz commented:
ENV VARIABLES
SECRET=rahsianiaga
GOOGLE_CLIENT_ID=40793720343-7bi7t406jrpt0cllabsfj68bk28o6s0g.apps.googleusercontent.com
GOOGLE_CLIENT_SECRET=Qfsx_VnNRnj0NWM3tT9YtuvQ
DEFAULT_SECRET=leviathan
